apps/studio/pages/api/ai/edge-function/complete.ts

```diff
@@ -0,0 +1,288 @@
+import { streamText } from 'ai'
+import { openai } from '@ai-sdk/openai'
+import { NextApiRequest, NextApiResponse } from 'next'
+import { getTools } from '../sql/tools'
+import { executeSql } from 'data/sql/execute-sql-query'
+import pgMeta from '@supabase/pg-meta'
+
+export const maxDuration = 30
+const openAiKey = process.env.OPENAI_API_KEY
+const pgMetaSchemasList = pgMeta.schemas.list()
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!openAiKey) {
+    return new Response(
+      JSON.stringify({
+        error: 'No OPENAI_API_KEY set. Create this environment variable to use AI features.',
+      }),
+      {
+        status: 500,
```

Comment:
we'll need to be consistent on what status to return for this - generate-v3 returns 400, here and generate-v2 returns 500

we probably should set all of them to 400?